### PR TITLE
Fixes #10241: "Package management" technique missing "allow untrusted" switch

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/10241

### CODE REVIEW

**Comment by amousset** [comment]
Do not merge for now, still testing.

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Ready for review!

**Comment by amousset** [comment]
Commit modified

File: techniques/applications/packageManagement/1.1/packageManagement.st [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+   
+  classes:
+      # This class is different from the one in ncf (where a version number and latest are considered "specified")
+      "version_specified_${index_pkg}" expression => strcmp("${version[${index_pkg}]}", "specific");
+      "architecture_specified_${index_pkg}" expression => strcmp("${architecture[${index_pkg}]}", "specific");
+      "posthook_specified_${index_pkg}" not => strcmp("${posthook[${index_pkg}]}", "");
+      "state_present_${index_pkg}" expression => strcmp("${state[${index_pkg}]}", "present");
+      "version_latest_${index_pkg}" expression => strcmp("${version[${index_pkg}]}", "latest");
+      "allow_unstrusted_${index_pkg}" expression => strcmp("${manager_allow_untrusted[${index_pkg}]}", "true");
+      "manager_apt_${index_pkg}" expression => strcmp("${manager[${index_pkg}]}", "apt");
```
Comment (amousset) [inline techniques/applications/packageManagement/1.1/packageManagement.st]: I wonder what to do here. If the user uses the default package manager, it won't pass the (apt-only) options.

Comment (amousset) [inline techniques/applications/packageManagement/1.1/packageManagement.st]: We could duplicate the logic for default package manager selection here, but I don't like it either.

Comment (amousset) [inline techniques/applications/packageManagement/1.1/packageManagement.st]: I will add the apt detection mechanism for now, and we will see if we need something else after

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Ready for review!

File: techniques/applications/packageManagement/1.1/packageManagement.st (hunk 1 above)
Comment (peckpeck) [inline techniques/applications/packageManagement/1.1/packageManagement.st]: Why not allow it for any package manager ?

Comment (peckpeck) [inline techniques/applications/packageManagement/1.1/packageManagement.st]: and detect apt only for the allow_untrusted part

Comment (amousset) [inline techniques/applications/packageManagement/1.1/packageManagement.st]: It's what is done now.

**Pull request merged**